src/writer/to_code.rs

```diff
@@ -145,26 +145,38 @@ impl ToCode for Chunk {
                 vec![s1, s2]
             }
             Name(ref name) => vec![name.clone()],
-            BoxFn { ref typ } => vec![format!("let user_data: Box<Box<{}>> = Box::new(Box::new(callback));", typ)],
-            ExternCFunc { ref name, ref parameters, ref body } => {
-                let prefix = format!(r#"unsafe extern "C" fn {}("#, name);
+            BoxFn { name: None, ref typ } => vec![
+                format!("let user_data: Box<Box<{}>> = Box::new(Box::new(callback));", typ),
+            ],
+            BoxFn { name: Some(ref name), ref typ } => vec![
+                format!("let {}_data: Box<Box<{}>> = Box::new(Box::new({}));",
```

Comment:
Why is `_data` appended here instead of just where the name was given?

Reply:
If it were this easy, it wouldn't be fun: I use the name as base to generate callbacks' name and variables.

Reply:
I see, that sounds more like a hack then though? I'm fine with keeping this if there's a comment explaining this and @EPashkin is also fine with it

Reply:
~IMHO this not hack, but prevention of duplicate names, Ex. we have parameter `page_func` have Box named `page_func_data` https://github.com/GuillaumeGomez/gtk/blob/new-gir/src/auto/assistant.rs#L241~
~We can use some fixed name instead, but it can conflict with parameters,
so I prefer current version.~
I was wrong, this case seems not used as I found only one `BoxFn` construction with `name:None` in code, and `page_func_data` added in `function_body_chunk.rs:add_trampoline`,
maybe `BoxFn` need refactoring: removing name or adding flags 'boxed', 'into' for use in `add_trampoline`

Reply:
I'll add a `TODO` comment.